### ISSUE TITLE
2.4.3 Focus Order - Submitting a form in error

### ISSUE THREAD

**Issue by anevins12** [w3c/wcag#3396]
Does 2.4.3 cover a scenario where submitting a form in error on a web page produces an error message, but keyboard focus is not moved to that error message? I'm thinking for users in the Benefits section of the Understanding doc, where adding a status message is insufficient.

Some folks say that this is not a failure because the error message is not focusable in the page navigation. I think this is down to the part of the criterion "If a Web page can be navigated sequentially." 

Is that really what is meant? To me, that reads if a webpage can be navigated sequentially, then carry on and read the next clause "and the navigation sequences affect meaning or operation", then finally reach the gist of the criterion "focusable components receive focus in an order that preserves meaning and operability."

**Comment by anevins12** [comment]
I think I got to the answer. No, 2.4.3 would not be applicable in this case. This criterion is only concerned with navigation sequences (i.e. TABing). 

Additionally, although the Understanding document of the criterion mentions the importance of moving keyboard focus for dialogs, the act of moving keyboard focus isn't necessarily what is required. The modal dialog could, alternatively, be positioned after the trigger control in the source code. What's important is that when users trigger a control to reveal a modal dialog, the modal dialog content is next in the TAB order. It's just that in reality modal dialogs are at the bottom of the DOM and difficult to move, or it's simply more predictable to move users for them and that's why it's recommended here.

**Comment by anevins12** [comment]
The only thing outstanding in the Understanding document is the statement about a screen magnifier user following the "focus order":

> Only a small portion of the page may be visible to an individual using a screen magnifier at a high level of magnification. Such a user may interpret a field in the wrong context if the focus order is not logical.

When we know some magnifier users have the option to follow the focus order (i.e. tracking keyboard focus) without entering a keystroke. 

Is it a mistake that the language used in the quote above does not use the term "keyboard"?

**Comment by fstrr** [comment]
This issue is labelled as a discussion, so we’re moving this to Discussions. There doesn’t seem to be an update to make to the documentation, but if that changes, we can move it back to the issues list.
